pricing-app/src/app/order/OrderBreakfastItem.tsx

```diff
@@ -0,0 +1,97 @@
+import { BagelTopping, BreakFast, SandwichTopping } from "@/types/enum";
+import { FC, useState } from "react";
+import SelectionBox from "../../components/selectionBox";
+import Options from "@/types";
+import { Item } from "@/types/type";
+
+const defaultBreakfastItem: Item = {
+  type: BreakFast.Sandwich,
+  topping: SandwichTopping.None,
+  quantity: 1,
+  cost: 0,
+};
+
+type Props = {
+  type: BreakFast;
+  toppingOptions: Options[];
+  onBreakfastItemAdded: (breakfast: Item) => void;
+};
+
+const OrderBreakfastItem: FC<Props> = ({
+  type,
+  toppingOptions,
+  onBreakfastItemAdded,
+}: Props) => {
+  const [breakfastItem, setBreakfastItem] = useState<Item>(defaultBreakfastItem);
+  const calculateCost = (): number => {
```

Comment:
Can we separate this logic out? ideally, each component should have a small single focus purpose.